components/src/core/InfoListItem/InfoListItem.tsx

```diff
@@ -128,46 +158,73 @@ const InfoListItemRender: React.ForwardRefRenderFunction<unknown, InfoListItemPr
     } = props;
 
     const combine = useCallback(
-        (className: keyof InfoListItemClasses): string => clsx(defaultClasses[className], classes[className]),
+        (className: keyof InfoListItemClasses): string => cx(defaultClasses[className], classes[className]),
         [defaultClasses, classes]
     );
 
     const getIcon = useCallback((): JSX.Element | undefined => {
         if (icon) {
             return (
                 <ListItemAvatar style={{ minWidth: 'unset' }}>
-                    <Avatar className={combine(avatar ? 'avatar' : 'icon')}>{icon}</Avatar>
+                    {avatar ? (
+                        <Icon
+                            statusColor={statusColor}
+                            iconColor={iconColor}
+                            avatar={avatar}
+                            iconAlign={iconAlign}
+                            className={combine('avatar')}
+                        >
+                            {icon}
+                        </Icon>
+                    ) : (
+                        <Icon
+                            statusColor={statusColor}
+                            iconColor={iconColor}
+                            avatar={avatar}
+                            iconAlign={iconAlign}
+                            className={combine('icon')}
+                        >
+                            {icon}
+                        </Icon>
+                    )}
```

Comment:
This is a lot of bloat...can we just put the conditional on the className prop instead since that's the only thing that's different between these two elements?